### ISSUE TITLE
Typescript fails to build due to glamorous type errors

### ISSUE THREAD

**Issue by jpandl19** [robinpowered/glamorous-native#102]
- `glamorous-native` version: 1.4.0
- `react-native` version: 0.55.4
- `react` version: 16.3.1

What you did: Tried to build my application via `tsc` 

What happened:
The following failures occured while trying to build the application.
```bash
node_modules/glamorous-native/typings/built-in-component-factories.d.ts:5:3 - error TS2724: Module '"/Users/jpandl/Code/personal/centsai_mobile/node_modules/@types/react-native/index"' has no exported member 'ScrollViewStyle'. Did you mean 'ScrollViewBase'?

5   ScrollViewStyle,
    ~~~~~~~~~~~~~~~

  node_modules/@types/react-native/index.d.ts:6420:15
    6420 declare const ScrollViewBase: Constructor<ScrollResponderMixin> & typeof ScrollViewComponent;
                       ~~~~~~~~~~~~~~
    'ScrollViewBase' is declared here.

node_modules/glamorous-native/typings/built-in-glamorous-components.d.ts:5:3 - error TS2724: Module '"/Users/jpandl/Code/personal/centsai_mobile/node_modules/@types/react-native/index"' has no exported member 'ScrollViewStyle'. Did you mean 'ScrollViewBase'?

5   ScrollViewStyle,
    ~~~~~~~~~~~~~~~

  node_modules/@types/react-native/index.d.ts:6420:15
    6420 declare const ScrollViewBase: Constructor<ScrollResponderMixin> & typeof ScrollViewComponent;
                       ~~~~~~~~~~~~~~
    'ScrollViewBase' is declared here.

node_modules/glamorous-native/typings/glamorous.d.ts:23:3 - error TS2724: Module '"/Users/jpandl/Code/personal/centsai_mobile/node_modules/@types/react-native/index"' has no exported member 'ScrollViewStyle'. Did you mean 'ScrollViewBase'?

23   ScrollViewStyle,
     ~~~~~~~~~~~~~~~

  node_modules/@types/react-native/index.d.ts:6420:15
    6420 declare const ScrollViewBase: Constructor<ScrollResponderMixin> & typeof ScrollViewComponent;
                       ~~~~~~~~~~~~~~
    'ScrollViewBase' is declared here.

node_modules/glamorous-native/typings/helpers.d.ts:4:55 - error TS2344: Type 'keyof T' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.

4 export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>

``` 

Reproduction:

Create a react-native application and setup typescript. Ensure `"skipLibCheck": false` Try to build the app. You should see the compiler errors.

Problem description:
There are two problms with the types here:
1) `ScrollViewStyle` does not exist in `react-native`.
2)  `...Diff<keyof T, K>>` also throws an error saying: `Type 'keyof T' does not satisfy the constraint 'string'`

Suggested solution:
1. Seems like we probably just want `ViewStyle` here, since that's what `ScrollView`'s `style` prop uses in `@types/react-native`: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-native/index.d.ts#L6410

2. `...Diff<string, string>>` satisfies the type constraint, but I'm not sure if that is the intended definition.

**Comment by jpandl19** [comment]
@epeli just took a look and approved the PR. Thanks!

**Comment by ajwhite** [comment]
#98 has been released under `v1.4.1`, thanks for the fix and the patience @epeli
